src/GraphQL/Types/DateGraphType.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Globalization;
+using Newtonsoft.Json.Linq;
+
+namespace GraphQL.Types
+{
+    public class DateGraphType : ScalarGraphType
```

Comment:
Would be nice to see some tests for the `Coerce` behavior.

One thing of note.  I don't really like having the direct dependency on JSON.NET here.  The way I did Input Objects in my branch was to convert all of those values up front.  That way you don't have to have JSON.NET littered everywhere and you only have to deal with core types.